CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+# Contributing
```

Comment:
Either here or in a future PR, it might be good to structure this doc around the different things people could contribute with and where to find them (e.g. improving docs vs adding new lints vs improving the test setup) and meta advice like "look for `E-mentor` and `E-help-wanted` labels."

What do you think?

Reply:
Yup, we could definitely use more stuff like that.  This was focused more on restructuring what we already had.